resources/galera_configuration.rb

```diff
@@ -58,9 +58,36 @@
   end
 end
 
+action :bootstrap do
+  ruby_block 'bootstrap galera cluster' do
+    block do
+      shell_out!('galera_new_cluster')
+      node.normal['mariadb']['galera']['bootstrapped'] = true # rubocop:disable ChefCorrectness/NodeNormal
+    end
+    notifies :stop, "service[#{platform_service_name}]", :before
+    not_if { galera_cluster_bootstrapped? && bootstrapped_attribute_set? }
+  end
+end
+
+action :join do
+  ruby_block 'join galera cluster' do
+    block do
+      node.normal['mariadb']['galera']['bootstrapped'] = true if galera_cluster_joined? # rubocop:disable ChefCorrectness/NodeNormal
+    end
+    notifies :restart, "service[#{platform_service_name}]", :before
+    not_if { galera_cluster_joined? && bootstrapped_attribute_set? }
+  end
+end
+
 action_class do
   include MariaDBCookbook::Helpers
 
+  def bootstrapped_attribute_set?
+    node['mariadb']['galera']['bootstrapped']
+  rescue
```

Comment:
The way that `Chef::Node::ImmutableMash` works I'm pretty sure this will return nil if **any** of the structure is missing? So no need for the rescue and this could just be a ternary or similar. If it was a plain Hash then it'd be better rescuing `NoMethodError` anyway.